lib/galaxy/tool_util/linters/xsd.py

```diff
@@ -0,0 +1,33 @@
+import os.path
+from typing import TYPE_CHECKING
+
+from galaxy import tool_util
+from galaxy.tool_util.lint import Linter
+from galaxy.util import etree
+
+if TYPE_CHECKING:
+    from galaxy.tool_util.lint import LintContext
+    from galaxy.tool_util.parser import ToolSource
+
+TOOL_XSD = os.path.join(os.path.dirname(tool_util.__file__), "xsd", "galaxy.xsd")
+
+
+class XSD(Linter):
+    """
+    Lint an XML tool against XSD
+    """
+
+    @classmethod
+    def lint(cls, tool_source: "ToolSource", lint_ctx: "LintContext"):
+        tool_xml = getattr(tool_source, "xml_tree", None)
+        if not tool_xml:
+            return
+        xmlschema = etree.parse(TOOL_XSD)
+        xsd = etree.XMLSchema(xmlschema)
+        if not xsd.validate(tool_xml):
+            for error in xsd.error_log:
```

Comment:
Not sure what to do with this error: 

```
lib/galaxy/tool_util/linters/xsd.py:28: error: "_ErrorLog" has no attribute
"__iter__" (not iterable)  [attr-defined]
                for error in xsd.error_log:
                             ^~~~~~~~~~~~~
```

As far as I can see it is iterable: https://github.com/lxml/lxml/blob/009e4ec30451ce370a42eefde91e03b5a8fbbab8/src/lxml/xmlerror.pxi#L474C9-L474C17

Reply:
Can you replace check if this passes with types-lxml instead of lxml-stubs ?

Reply:
Not sure if this still applies https://github.com/abelcheung/types-lxml?tab=readme-ov-file#important-note

Reply:
> Can you check if this passes with types-lxml instead of lxml-stubs ?

No ... new problems. Also no idea how to fix.